### ISSUE TITLE
error TS4053: Return type of public method from exported class has or is using name 'Response' from external module "/node_modules/@angular/http/src/static_response" but cannot be named.

### ISSUE THREAD

**Issue by klochko7** [driftyco/ionic#8432]
Hi everyone, 
I tried to update my project with Ionic2 RC0. I choose the way creating a new Ionic 2 project and copying my project files to it. What can I do to solve this ? The first line with error..

```
 login(user:User) {
     const data = JSON.stringify({u: user.username, p: user.password});
     return this.http.post(this.actions.login, data)
 }
```

**Comment by manucorporat** [comment]
Maybe by:
```
 login(user:User): Response {
     const data = JSON.stringify({u: user.username, p: user.password});
     return this.http.post(this.actions.login, data)
 }
```

Idk, I can't reproduce it.

**Comment by jgw96** [comment]
Hello @klochko7 ! Thanks for using Ionic! This is actually a typescript error and not something caused by the framework. Because of this I am going to close this issue and ask that you ask this question on the [forum](https://forum.ionicframework.com).

**Comment by kinggolf** [comment]
Hi @klochko7 and @jgw96 - ran into same ts error on http.post code when upgrading to ionic-angular 2.0.0-rc.0. Same exact code is used in 2.0.0-beta.11 version w/ no ts error and in a separate Angular 2 app using @angular 2.0.0 and build w/ angular-cli .0.0-beta.16. After digging around for awhile, I noticed diffs in tsconfig.json files pertaining to declarations under compilerOptions:
- upgraded Ionic2 app w/ ts errors had declaration set to true
- previous version of same Ionic2 app had no declaration parameter (default false?)
- separate NG2 app had declaration set to false
When I changed the declaration compilerOption to false in my upgraded Ionic2 app, then the ts errors were gone. 

Not disputing @jgw96 pointing out this is Typescript issue, but I wanted to share my findings. My understanding of the declarations parameter under compilerOptions is related to generating ts definition files, which I certainly do not need in my app. Further comments/thoughts welcome.

Thanks Ionic! Cheers.
